### PR TITLE
Feature/additional services

### PR DESCRIPTION
**Pull request by emarf**
Added xray implementation and maintainer for each test
Screenshot 2022-01-14 at 17 29 45
Screenshot 2022-01-14 at 17 30 19
Screenshot 2022-01-14 at 17 28 58
Screenshot 2022-01-14 at 17 28 38

Xray has 4 params: 
- `xrayExecutionKey` - mandatory (use beforEach hook)
- `xrayTestKey`- mandatory (use in test)
- `disableSync` - optional (use beforEach hook)
- `enableRealTimeSync` - optional (use beforEach hook)

Maintainer use in test (if not announced `anonymous`).

Problems: 
beforeEach hook not very practical, better to use beforeAll, but it doesn't work like in the documentation.
if you find solution, contact me pls;)

Right now i am implementing testrail and zephyr tcm in this branch

### CODE REVIEW

**Comment by ryanrosello-og** [comment]
hey @emarf, this looks pretty nice.

 prolly need this issued fixed https://github.com/ryanrosello-og/zebrunner-playwright-agent/issues/22 though before this PR can be merged

**Comment by emarf** [comment]
@ryanrosello-og could you share for me test from #22, because with your tests everything okay
Screenshot 2022-01-16 at 16 13 53
Screenshot 2022-01-16 at 16 14 04
Screenshot 2022-01-16 at 16 14 34
Screenshot 2022-01-16 at 16 14 42

**Comment by emarf** [comment]
TestRail
- `testRailSuiteId`(string) - `mandatory` use beforeEach hook,
- `testRailCaseId`(array of string) - `mandatory` use in the test,
- `testRailRunId`(string) - `optional` use beforeEach hook,
- `testRailRunName`(string) - `optional` use beforeEach hook,
- `testRailMilestone`(string) - `optional` use beforeEach hook,
- `testRailAssignee`(string) - `optional` use beforeEach hook,
- `testRailDisableSync`(string, but need to be true or false) - `optional` use beforeEach hook,
- `testRailIncludeAll`(string, but need to be true or false) - `optional` use beforeEach hook,
- `testRailEnableRealTimeSync`(string, but need to be true or false) - `optional` use beforeEach hook,  if value `true` `testRailIncludeAll` also `true`

Xray
- `xrayExecutionKey`(string) - `mandatory` use beforeEach hook,
- `xrayTestKey`(array of string) - `mandatory` use in the test,
- `xrayDisableSync`(string, but need to be true or false) - `optional` use beforeEach hook,
- `xrayEnableRealTimeSync`(string, but need to be true or false) - `optional` use beforeEach hook,

Zephyr
- `zephyrTestCycleKey`(string) - `mandatory` use beforeEach hook,
- `zephyrJiraProjectKey`(string) - `mandatory` use beforeEach hook
- `zephyrTestCaseKey`(array of string) - `mandatory` use in the test,
- `zephyrDisableSync`(string, but need to be true or false) - `optional` use beforeEach hook,
- `zephyrEnableRealTimeSync`(string, but need to be true or false) - `optional` use beforeEach hook,

File: playwright.config.ts [hunk 1]
```diff
@@ -23,11 +23,12 @@ const config: PlaywrightTestConfig = {
       use: {...devices['Desktop Safari']},
     },
   ],
+  testMatch:['pw_nested_testsuite.spec.ts'],
```
Comment (emarf) [inline playwright.config.ts]: need to remove

**Comment by emarf** [comment]
Changed tcm implementation and rewrite through `process.sdout.write()`
Can you look on code and give your review please?

**Review by ryanrosello-og — APPROVED** [review]
LGTM

**Comment by ryanrosello-og** [comment]
Probably best to update the README at some point but that could be done in a separate PR.

**Pull request merged**